src/test/java/com/redhat/devtools/lsp4ij/fixtures/LSPCompletionClientConfigFixtureTestCase.java

```diff
@@ -0,0 +1,155 @@
+     * accepting the first completion. <code>fileBodyBefore</code> should include a <code>&lt;caret&gt;</code>
+     * placeholder for where completion should be triggered, and <code>fileBodyAfter</code> should include either
+     * a <code>&lt;template&gt;</code> placeholder or a <code>&lt;caret&gt;</code> placeholder based on whether or
+     * not a template or just the caret is expected at the offset corresponding to that placeholder.
+     *
+     * @param fileName                      the file name
+     * @param fileBodyBefore                the file body before including the embedded typing imperative comment
+     * @param fileBodyAfter                 the file body after the character has been typed and formatting applied
+     * @param mockTextCompletionJson        the mock text/completion JSON response
+     * @param mockCompletionItemResolveJson the mock completionItem/resolve JSON response
+     * @param clientConfigCustomizer        a function that allows the caller to customize the client configuration as
+     *                                      needed for the test scenario
+     */
+    protected void assertTemplateForArguments(@NotNull String fileName,
+                                              @NotNull String fileBodyBefore,
+                                              @NotNull String fileBodyAfter,
+                                              @NotNull String mockTextCompletionJson,
+                                              @NotNull String mockCompletionItemResolveJson,
+                                              @Nullable Consumer<ClientConfigurationSettings> clientConfigCustomizer) {
+        MockLanguageServer.INSTANCE.setTimeToProceedQueries(200);
+
+        CompletionList mockCompletionList = JSONUtils.getLsp4jGson().fromJson(mockTextCompletionJson, CompletionList.class);
+        MockLanguageServer.INSTANCE.setCompletionList(mockCompletionList);
+
+        CompletionItem mockCompletionItem = JSONUtils.getLsp4jGson().fromJson(mockCompletionItemResolveJson, CompletionItem.class);
+        MockLanguageServer.INSTANCE.setCompletionItem(mockCompletionItem);
+
+        Project project = myFixture.getProject();
+        PsiFile file = myFixture.configureByText(fileName, removeCaret(fileBodyBefore));
+        Editor editor = myFixture.getEditor();
+        CaretModel caretModel = editor.getCaretModel();
+
+        // Initialize the language server
+        List<LanguageServerItem> languageServers = new LinkedList<>();
+        try {
+            ContainerUtil.addAllNotNull(languageServers, LanguageServiceAccessor.getInstance(project)
+                    .getLanguageServers(file.getVirtualFile(), null, null)
+                    .get(5000, TimeUnit.MILLISECONDS));
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+
+        // Configure the language server's completion client configuration
+        LanguageServerItem languageServer = ContainerUtil.getFirstItem(languageServers);
```

Comment:
Now we need to get the language server that's going to be used and set it up properly for client config testing.